rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/telemetry/TbMsgDeleteAttributesTest.java

```diff
@@ -0,0 +1,119 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@Slf4j
+public class TbMsgDeleteAttributesTest {
+    final ObjectMapper mapper = new ObjectMapper();
+
+    DeviceId deviceId;
+    TbMsgDeleteAttributes node;
+    TbMsgDeleteAttributesConfiguration config;
+    TbNodeConfiguration nodeConfiguration;
+    TbContext ctx;
+    TbMsgCallback callback;
+
+    RuleEngineTelemetryService telemetryService;
+
+    @BeforeEach
+    void setUp() throws TbNodeException {
+        deviceId = new DeviceId(UUID.randomUUID());
+        callback = mock(TbMsgCallback.class);
+        ctx = mock(TbContext.class);
+        config = new TbMsgDeleteAttributesConfiguration().defaultConfiguration();
+        config.setKeys(List.of("${TestAttribute_1}", "$[TestAttribute_2]", "$[TestAttribute_3]", "TestAttribute_4"));
+        nodeConfiguration = new TbNodeConfiguration(mapper.valueToTree(config));
+        node = spy(new TbMsgDeleteAttributes());
+        node.init(ctx, nodeConfiguration);
+        telemetryService = mock(RuleEngineTelemetryService.class);
+
+        willReturn(telemetryService).given(ctx).getTelemetryService();
+        willAnswer(invocation -> {
+            TelemetryNodeCallback callBack = invocation.getArgument(4);
+            callBack.onSuccess(null);
+            return null;
+        }).given(telemetryService).deleteAndNotify(
+                any(), any(), anyString(), anyList(), any());
+    }
+
+    @AfterEach
+    void tearDown() {
+        node.destroy();
+    }
+
+    @Test
+    void givenDefaultConfig_whenInit_thenOK() {
+        assertThat(node.config).isEqualTo(config);
+    }
+
+    @Test
+    void givenDefaultConfig_whenVerify_thenOK() {
+        TbMsgDeleteAttributesConfiguration defaultConfig = new TbMsgDeleteAttributesConfiguration().defaultConfiguration();
+        assertThat(defaultConfig.getScope()).isEqualTo(DataConstants.SERVER_SCOPE);
+        assertThat(defaultConfig.getKeys()).isEqualTo(Collections.emptyList());
+    }
+
+    @Test
+    void givenMsg_whenOnMsg_thenVerifyOutput() throws Exception {
+        final Map<String, String> mdMap = Map.of(
+                "TestAttribute_1", "temperature",
+                "city", "NY"
+        );
+        final TbMsgMetaData metaData = new TbMsgMetaData(mdMap);
+        final String data = "{\"TestAttribute_2\": \"humidity\", \"TestAttribute_3\": \"voltage\"}";
+
+        TbMsg msg = TbMsg.newMsg("POST_ATTRIBUTES_REQUEST", deviceId, metaData, data, callback);
+        node.onMsg(ctx, msg);
+
+        ArgumentCaptor<TbMsg> newMsgCaptor = ArgumentCaptor.forClass(TbMsg.class);
+        verify(ctx, times(1)).tellSuccess(newMsgCaptor.capture());
+        verify(ctx, never()).tellFailure(any(), any());
+        verify(telemetryService, times(1)).deleteAndNotify(any(), any(), anyString(), anyList(), any());
+    }
+}
```

Comment:
Please, add a new line at the end of the file, and new line after givenMsg_whenOnMsg_thenVerifyOutput method